File: 75/1.py
from typing import List
from collections import Counter


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        index = 0

        counter = Counter(nums)

        for wtf in range(3):
            key = wtf
            value = counter[key]

            for i in range(index, index + value):
                nums[i] = key
            index += value
        return


sln = Solution()
nums = [2, 0, 2, 1, 1, 0]
sln.sortColors(nums)
print(nums)

File: 45/1.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
  int jump(vector<int>& nums)
  {
    auto steps = vector<int>(nums.size());
    steps[steps.size() - 1] = 0;

    for (int index = steps.size() - 1; --index >= 0;) {
      if (nums[index] + index + 1 >= nums.size()) {
        steps[index] = 1;
      } else {
        int min = INT_MAX;
        for (int i = index + nums[index]; i > index; --i) {
          if (steps[i] < min) {
            min = steps[i];
          }
        }
        if (min != INT_MAX) {
          ++min;
        }
        steps[index] = min;
      }
    }

    return steps[0];
  }
};

int
main()
{
  auto sln = Solution();
  auto nums = vector<int>{ 2, 3, 0, 1, 4 };

  cout << sln.jump(nums) << endl;
  getchar();
}
